src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -353,15 +393,19 @@ bool CreateJavaSources (IEnumerable<TypeDefinition> javaTypes, TypeDefinitionCac
 
 				using (var writer = MemoryStreamPool.Shared.CreateStreamWriter ()) {
 					try {
-						var jti = new JavaCallableWrapperGenerator (t, Log.LogWarning, cache) {
+						var jti = new JavaCallableWrapperGenerator (t, Log.LogWarning, cache, classifier) {
 							GenerateOnCreateOverrides = generateOnCreateOverrides,
 							ApplicationJavaClass = ApplicationJavaClass,
 							MonoRuntimeInitialization = monoInit,
 						};
 
 						jti.Generate (writer);
 #if ENABLE_MARSHAL_METHODS
-						overriddenMethodDescriptors.AddRange (jti.OverriddenMethodDescriptors);
+						if (!Debug) {
```

Comment:
We could remove this `if` + indentation if we just did `if (classifier?.FoundDynamicallyRegisteredMethods)`, thus implicitly relying on the fact that `classifier` is null in Debug builds.

But that means relying on "implicit context".

Not sure which tradeoff is "better": minimizing indentation or keeping things explicit.

Reply:
I much prefer if the code is explicit, makes it easier to understand the flow for someone not familiar with it